scripts/install.sh

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+set -e
+
+# Determine the platform using 'OS' and 'ARCH'
+initPlatform() {
+    OS=$(uname -s | tr '[:upper:]' '[:lower:]')
+    ARCH=$(uname -m)
+    if [ $OS == "darwin" ] && [ $ARCH == "arm64" ]; then
+        PLATFORM="darwin_arm64"
+    elif [ $OS == "darwin" ] && [ $ARCH == "x86_64" ]; then
+        PLATFORM="darwin_amd64"
+    elif [ $OS == "linux" ] && [ $ARCH == "x86_64" ]; then
+        PLATFORM="linux_amd64"
+    else
+        printf "Platform not supported: os=$OS arch=$ARCH\n"
+        exit 1
+    fi
+}
+
+# Verify that a binary is available for the current combination of 'PLATFORM' and 'VERSION'
+verifyAvailability() {
+    if [ $VERSION == nightly ] && [ $PLATFORM == macos-arm64 ]; then
+        printf "\nNightly builds are currently not published for ${PLATFORM}.\n\n"
+        read -p "Do you want to install the nightly macos-x64 build which can run using Rosetta 2 instead? [y/N]" -n 1 -r -s < /dev/tty
+        printf "\n\n"
+        if [[ ! $REPLY =~ ^[Yy]$ ]]; then
+            exit 0
+        fi
+        PLATFORM=macos-x64
+    fi
+}
+
+# Create a temporary directory and setup deletion on script exit using the 'EXIT' signal
+initTmpDir() {
+    TMP_DIR=$(mktemp -d)
+    trap 'rm -rf -- "${TMP_DIR}"' EXIT
+    cd $TMP_DIR
+}
+
+# Download the binary and check the integrity using the SHA256 checksum
+downloadBinary() {
+    CDN="cdn.rilldata.com"
+    BINARY_URL="https://${CDN}/rill/${VERSION}/rill_${PLATFORM}.zip"
+    CHECKSUM_URL="https://${CDN}/rill/${VERSION}/checksums.txt"
+
+    printf "Downloading binary: ${BINARY_URL}\n"
+    curl --location --progress-bar "${BINARY_URL}" --output rill_${PLATFORM}.zip
+
+    printf "\nDownloading checksum: ${CHECKSUM_URL}\n"
+    curl --location --progress-bar "${CHECKSUM_URL}" --output checksums.txt
+
+    printf "\nVerifying the SHA256 checksum of the downloaded binary:\n"
+    shasum --algorithm 256 --ignore-missing --check checksums.txt
+
+    printf "\nUnpacking rill_${PLATFORM}.zip\n"
+    unzip rill_${PLATFORM}.zip
```

Comment:
I’m a bit worried about `unzip`, it’s not installed by default on many Linux distros. Should we create a small pre-install test in the script that checks the needed tools (`curl` , `unzip` , `shasum` ) and print something helpful if they are not installed?

Reply:
@begelundmuller any thoughts on this one?

Reply:
I think checking that `unzip` and `shasum` exist on the system and failing with a friendly error message sounds like a good solution. People already need `curl` to get the install script.